File: testdata/bench/go/BenchmarkTest00067.go
package bench

import (
	"io"
	"net/http"
)

// SSRF via string concat URL
func Handler00067(w http.ResponseWriter, r *http.Request) {
	host := r.URL.Query().Get("host")
	target := "http://" + host + "/api/data"
	resp, _ := http.Get(target)
	defer resp.Body.Close()
	io.Copy(w, resp.Body)
}

File: testdata/bench/rust/BenchmarkTest00067.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: SSRF via partial URL construction
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("endpoint").unwrap_or(&String::new()).clone();
    let url = format!("http://internal-api/{}", param);
    let resp = reqwest::get(&url).await.unwrap();
    HttpResponse::Ok().body(resp.text().await.unwrap_or_default())
}
